ansible/roles/ocp4-workload-authentication/tasks/workload.yml

```diff
@@ -1,48 +1,64 @@
 ---
-# Implement your Workload deployment tasks here
-
-- name: Check that ocp4_idm_install is defined
+- name: Set up ocp4_workload_authentication combined dictionary
+  set_fact:
+    ocp4_workload_authentication: >-
+      {{ ocp4_workload_authentication_defaults
+       | combine(ocp4_workload_authentication_input  | default( {} ),
+                 ocp4_workload_authentication_secret | default( {} ))
+      }}
+- name: Print combined role variables
+  debug: var=ocp4_workload_authentication
+
+- name: Check that ocp4_workload_authentication.idm_type is defined
   when:
-  - ocp4_idm_install is not defined
+  - ocp4_workload_authentication.idm_type is not defined
   fail:
-    msg: "ocp4_idm_install is not defined"
+    msg: "ocp4_workload_authentication.idm_type is not defined"
 
 - name: Setup HTPasswd Authentication
-  when: ocp4_idm_install == "htpasswd"
+  when: ocp4_workload_authentication.idm_type == "htpasswd"
```

Comment:
FYI i find this way convenient for managing assertions / checks: https://github.com/redhat-gpte-devopsautomation/babylon_aws_sandbox/blob/master/tasks/pre_checks.yaml

Reply:
Changed it to assert.